Model/Connection.php

```diff
@@ -55,7 +55,10 @@ public function connect()
         if ($this->connection === false) {
             $this->state = self::STATE_INVALID_ADDRESS;
         } else {
-            $auth = @ssh2_auth_password($this->connection, $profile->getLogin(), $profile->getPassword());
+            $auth = @ssh2_auth_pubkey_file($this->connection, $profile->getLogin(), $profile->getRsaPubKey(), $profile->getRsaPemKey(), $profile->getPassphrase());
```

Comment:
Maybe do it only when a RsaPubKey is specified, instead of try to connect with an empty Rsa, then connect without Rsa ?